PyTDF/Node.py

```diff
@@ -55,20 +54,29 @@ def _dfs(self, node, prev_object= None):
             self._dfs(n, node.value)
 
     
-    def _graph_prune_util(self, node):
+    def _graph_prune(self):
 
         children = []
 
-        for n in node.next_nodes:
-            if self._graph_prune_util(n):
+        for n in self.next_nodes:
+            if n._graph_prune():
                 children.append(n)
-        node.next_nodes = children
+        self.next_nodes = children
+
+        if not self.next_nodes and (len(gc.get_referrers(self)) <= 3):
```

Comment:
You can also remove the parenthesis of the second condition in this `and`: `and len(gc...`